apartmenttherapy/order-service
# Elaine/prevent dupes

#### What does this PR do & why?
Preventing duplicates in a couple of places: 
1. A buyer may not create more than one open order for a product; and
2. A seller may not accept more than one order for a product.

#### How can this be manually tested?
- Tests should run.
- Verify a buyer cannot create a second order for a product, when one already exists that's `PENDING_ACCEPTANCE`, `ACCEPTED`, or `PAID`.
- Verify a buyer can create a second order for a product, when one exists that's `CLOSED` (for any reason).
- Verify a seller cannot accept a second order for a product, when one already exists that's `ACCEPTED` or `PAID`.
- Verify a seller can accept a second order for a product, when one exists that's `PENDING_ACCEPTANCE` or `CLOSED` (for any reason).

#### Any questions or concerns about this change?
*(Any significant changes you're introducing? What side effects might this have? What other parts of the app could this affect?)*


#### Deployment Notes
*(e.g. Does this require a migration or special infrastructure change? Clearing the cache? Third party service?)*


#### Link to story card and other documentation
https://app.asana.com/0/694906336753887/1113320261512505

#### GIF


#### Have you remembered to...?
